docs/reference/modules/cluster/disk_allocator.asciidoc

```diff
@@ -2,9 +2,60 @@
 ==== Disk-based shard allocation settings
 [[disk-based-shard-allocation-description]]
 // tag::disk-based-shard-allocation-description-tag[]
-{es} considers the available disk space on a node before deciding
-whether to allocate new shards to that node or to actively relocate shards away
-from that node.
+
+The <<shards-rebalancing-settings,balance>> of the cluster depends only on the
+number of shards on each node and the indices to which those shards belong. It
+considers neither the sizes of these shards nor the available disk space on
+each node, for the following reasons:
```

Comment:
I wonder if mixing in the concept of balance here is more confusing than not. We may want to just start with your `The disk-based shard allocator...` paragraph. The gist of this text seems adequately covered there and in the later admon about unequal disk usage.

Reply:
Hmm, good point about leading with the later paragraph. I'll think about removing this vs moving it elsewhere. It's a common source of confusion that "balanced" doesn't mean "equal disk usage", I think we need to spell out why that isn't the case. But there's no need to lead with this.